web/src/components/watches/DownstreamWatchVersionDiff.tsx

```diff
@@ -8,23 +8,43 @@ import DiffEditor from "../shared/DiffEditor";
 import "../../scss/components/watches/DownstreamWatchVersionDiff.scss";
 import Icon from "../Icon";
 
-class DownstreamWatchVersionDiff extends React.Component {
-  constructor() {
-    super();
+type Props = {
+  firstSequence: number;
+  hideBackButton?: boolean;
+  onBackClick: (goBack: boolean) => void;
+  secondSequence: number;
+  slug: string;
+} & RouteComponentProps;
+
+type State = {
+  loadingFileTrees: boolean;
+  firstApplicationTree?: ApplicationTree;
+  secondApplicationTree?: ApplicationTree;
+  fileLoading: boolean;
+  fileLoadErr: boolean;
+  fileLoadErrMessage?: string;
+  hasErrSettingDiff: boolean;
+  errSettingDiff?: string;
+  failedSequence?: number;
+};
+
+type ApplicationTree = {
+  files: {
+    [key: string]: string;
+  };
+};
+class DownstreamWatchVersionDiff extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
     this.state = {
       loadingFileTrees: false,
-      firstApplicationTree: {},
-      secondApplicationTree: {},
       fileLoading: false,
       fileLoadErr: false,
-      fileLoadErrMessage: "",
       hasErrSettingDiff: false,
-      errSettingDiff: "",
-      failedSequence: undefined,
```

Comment:
I don't think these default values are necessary and complicate the type definitions